### PR TITLE
Fixed Debounce in `AddClaimantModal`

### PR DESCRIPTION
**Pull request by jcq**
Connects #14430

### Description
Fixes regression from #14202; this brings fully functioning debounce functionality in `AddClaimantModal` along with additional JS tests

### Acceptance Criteria
- [ ] Code compiles correctly
- [ ] API calls are properly debounced in `AddClaimantModal`

### CODE REVIEW

File: client/app/intake/components/AddClaimantModal.jsx [hunk 1]
```diff
@@ -1,31 +1,49 @@
...
-];
+const relationshipOpts = [{ label: 'Attorney (previously or currently)', value: 'attorney' }];
 
-export const AddClaimantModal = ({ onCancel, onSubmit }) => {
+const fetchAttorneys = async (search = '') => {
+  const res = await ApiUtil.get('/intake/attorneys', { query: { query: search } });
+
+  return res?.body;
+};
+const getClaimantOpts = async (search = '', asyncFn) => {
```
Comment (leikkisa) [inline client/app/intake/components/AddClaimantModal.jsx:17]: This looks nicer!

File: client/app/intake/components/AddClaimantModal.jsx [hunk 2]
```diff
@@ -1,31 +1,49 @@
...
 
-  useEffect(() => {
-    setDropdownOpts(
-      searchResults.map((item) => ({
-        label: item.fullName,
-        value: item.participantId
-      }))
-    );
-  }, [searchResults]);
+  const asyncFn = useCallback(
```
Comment (leikkisa) [inline client/app/intake/components/AddClaimantModal.jsx:41]: This looks like the main fix...

File: client/test/app/intake/components/AddClaimantModal.test.js [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+      await userEvent.type(input, 'BC');
+      jest.advanceTimersByTime(DEBOUNCE);
+      expect(func).toHaveBeenCalled();
+      expect(func).toBeCalledTimes(1);
+      await userEvent.type(input, 'DEF');
+      expect(func).toBeCalledTimes(1);
+      await userEvent.type(input, 'GHI');
+      expect(func).toBeCalledTimes(1);
+      jest.advanceTimersByTime(DEBOUNCE);
+      expect(func).toBeCalledTimes(2);
```
[the remark this answers is not in the document]
Comment (nanotone) [reply client/test/app/intake/components/AddClaimantModal.test.js:105]: Yeah, I think this test is great, and could use two more expectations to really cover everything:
- expect no function calls immediately after `BC` has been typed
- expect function calls to stay at 2 after one extra debounce period, as Sally suggested

File: client/app/intake/components/SelectClaimant.jsx [hunk 4]
```diff
@@ -34,6 +34,22 @@ const noClaimantsCopy = React.createElement(
   COPY.CLAIMANT_NOT_FOUND_END
 );
 
+const RemovableRadioLabel = ({ text, onRemove }) => (
+  <>
+    <span>{text}</span>{' '}
+    {onRemove && (
```
[the remark this answers is not in the document]
Comment (jcq) [reply client/app/intake/components/SelectClaimant.jsx:40]: Basically just a check to ensure that we don't display a button that doesn't do anything. It should be moot due to the way that we're using it, but it was just for covering bases.

**Pull request merged**